### PR TITLE
Swizzle a selector in all super classes

### PR DESCRIPTION
**Pull request by erychagov**

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m [hunk 1]
```diff
@@ -116,7 +116,15 @@ + (IMP)swizzleOriginalSelector:(SEL)originalSelector withCustomSelector:(SEL)cus
   IMP customImp = class_getMethodImplementation(self, customSelector);
   IMP originalImp = NULL;
   BOOL methodAdded = NO;
-  
+
+  // Replace implementation in super classes
+  Class baseClass = originalClass.superclass;
+  if( baseClass != [NSObject class]) {
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m]: I agree the first comment that you need to check if it is implemented or not.
BTW for the second part, isn't NSObject always be the root class of other classes in Objective C? Just curios.

Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m]: @jaelim-ms in theory: no, in practice: yes. Check out everything that Mike Ash writes about ObjC/Swift, e.g. his [blog on NSobject](https://www.mikeash.com/pyblog/friday-qa-2013-10-25-nsobject-the-class-and-the-protocol.html).

File: MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m [hunk 2]
```diff
@@ -116,7 +116,15 @@ + (IMP)swizzleOriginalSelector:(SEL)originalSelector withCustomSelector:(SEL)cus
   IMP customImp = class_getMethodImplementation(self, customSelector);
   IMP originalImp = NULL;
   BOOL methodAdded = NO;
-  
+
+  // Replace implementation in super classes
+  Class baseClass = originalClass.superclass;
```
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m]: Can we be sure that originalClass actually responds to superclass? In practice: yes, in theory: no. So we should check that originalClass actually responds to `superclass`.

Or am I too paranoid @jaelim-ms @clpolet ?

Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m]: I don't think I understand, @TroubleMakerBen do you have examples in mind?

File: MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m (hunk 1 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Application Delegate/MSAppDelegateForwarder.m]: Right NSObject is not always the root class but it's most probably is, though it doesn't hurt the way it is right now I think. To me it's maybe more an optimization.

**Comment by clpolet** [comment]
We figured out that those scenario were already mostly covered by the current implementation. Will do another PR.

**Pull request closed without merging**